mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -185,4 +198,20 @@ private Properties getCatalogProperties(org.apache.hadoop.hive.metastore.api.Tab
 
     return properties;
   }
+
+  private Schema schema(Properties properties, org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (properties.getProperty(InputFormatConfig.TABLE_SCHEMA) != null) {
+      return SchemaParser.fromJson(properties.getProperty(InputFormatConfig.TABLE_SCHEMA));
```

Comment:
In this case, I think we also need to check the hmsTable schema and validate that it is either null or compatible with the one set in the table property. I'd prefer requiring it to be null.

Reply:
Good point. Added the compatibility check here as well.